external-builds/pytorch/run_linux_pytorch_tests.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+"""Runs pytest for PyTorch unit tests with additional test exclusion
+
+Test exclusion can be due to
+- amdgpu family
+- pytorch version
+- general failures not yet upstream
+
+====================================
+EXPECTED INPUT ENVIRONMENT VARIABLES
+====================================
+INPUT_THEROCK_ROOT_DIR (optional)  - To change the root directory of TheRock
+                                   - Otherwise the location is extrapolated from
+                                   - the location of this script
+INPUT_AMDGPU_FAMILY    (optional)  - amdgpu_family as to run the tests on
+                                   - names used as in "TheRock/cmake/therock_amdgpu_targets.cmake"
+                                   - if not set, can lead to test failure as not enough tests might be omitted
+INPUT_PYTORCH_VERSION  (optional)  - PyTorch version used for the tests
+                                   - major.minor version as a string (e.g. "2.10")
+                                   - if not set: auto-detection based on pytorch/version.txt
+====================================
+"""
+
+import sys
+import os
+import tempfile
+
+from skip_tests.create_skip_tests import *
+import pytest
+
+
+def setup_env(pytorch_dir):
+    os.environ["PYTORCH_PRINT_REPRO_ON_FAILURE"] = "0"
+    os.environ["PYTORCH_TEST_WITH_ROCM"] = "1"
+    os.environ["MIOPEN_CUSTOM_CACHE_DIR"] = tempfile.mkdtemp()
+    os.environ["PYTORCH_TESTING_DEVICE_ONLY_FOR"] = "cuda"
+    old_pythonpath = os.getenv("PYTHONPATH", "")
+    test_dir = f"{pytorch_dir}/test"
+    if old_pythonpath != "":
+        os.environ["PYTHONPATH"] = f"{test_dir}:{old_pythonpath}"
+    else:
+        os.environ["PYTHONPATH"] = f"{test_dir}"
+    # we need to force update the PYTHONPATH to the sys path
+    # otherwise our current python process that will run
+    # pytest will not find it and pytest will crash!
+    if test_dir not in sys.path:
+        sys.path.insert(0, test_dir)
+
+
+if __name__ == "__main__":
+    root_dir = os.getenv("INPUT_THEROCK_ROOT_DIR", "")
+    # autodetect root dir via path of the script
+    if root_dir == "":
+        script_dir = os.path.dirname(sys.argv[0])
+        # we are in <TheRock Root Dir>/external-builds/pytorch
+        root_dir = script_dir.rsplit("/", 2)[0]
+
+    pytorch_dir = f"{root_dir}/external-builds/pytorch/pytorch"
+
+    amdgpu_family = os.getenv("INPUT_AMDGPU_FAMILY", "")
+    pytorch_version = os.getenv("INPUT_PYTORCH_VERSION", "")
+
+    # auto detect version by reading version string from pytorch/version.txt
+    if pytorch_version == "":
+        with open(f"{pytorch_dir}/version.txt", "r") as file:
+            line = file.read()
+            # hopefully pytorch version always stays like
+            # major.minor.<some more version stuff without dots>
+            pytorch_version = line.rsplit(".", 1)[0].strip()
```

Comment:
You can parse this from the torch package itself using https://packaging.pypa.io/en/stable/version.html

You can get the version string using
* The more modern approach (since Python 3.8): https://docs.python.org/3.13/library/importlib.metadata.html

    ```python
    >>> from importlib.metadata import version
    >>> version("torch")
    '2.10.0a0+rocmsdk20250919'
    ```

* The older approach: https://stackoverflow.com/a/20180564 (which imports the package -- that is slow for torch)

See also https://stackoverflow.com/a/17767076